extmod/modlwip.c

```diff
@@ -746,19 +781,25 @@ STATIC mp_obj_t lwip_socket_accept(mp_obj_t self_in) {
     }
 
     // accept incoming connection
-    if (socket->incoming.connection == NULL) {
+    struct tcp_pcb *volatile *incoming_connection;
```

Comment:
It's structure should be volatile, not pointer to it.

Reply:
This was what it took to appease the compiler.  It's the pointer `*incoming_connection* that is volatile and can change (ie change from NULL to something non-NULL).